skimage/future/graph/rag.py

```diff
@@ -263,6 +266,125 @@ def rag_mean_color(image, labels, connectivity=2, mode='distance',
            http://citeseerx.ist.psu.edu/viewdoc/summary?doi=10.1.1.11.5274
 
     """
+
+
+    def define_graph(graph,labels,image, extra_arguments=[],
```

Comment:
Oh I see, but nested functions are generally bad for performance. Cause even though the function is doing something specific, it has to be instantiated every time you made a call to the parent function. What I would do is put this outside with a name like `define_mean_color_rag` or something similar.